sdk/include/opentelemetry/sdk/common/attributemap_hash.h

```diff
@@ -57,6 +56,37 @@ inline size_t GetHashForAttributeMap(const OrderedAttributeMap &attribute_map)
   return seed;
 }
 
+// Calculate hash of keys and values of KeyValueIterable, filtered using callback.
+inline size_t GetHashForAttributeMap(
+    const opentelemetry::common::KeyValueIterable &attributes,
+    nostd::function_ref<bool(nostd::string_view)> is_key_present_callback)
+{
+  AttributeConverter converter;
+  size_t seed = 0UL;
+  attributes.ForEachKeyValue(
+      [&](nostd::string_view key, opentelemetry::common::AttributeValue value) noexcept {
+        if (!is_key_present_callback(key))
+        {
+          return true;
+        }
+        std::hash<std::string> hasher;
+        // reference -
+        // https://www.boost.org/doc/libs/1_37_0/doc/html/hash/reference.html#boost.hash_combine
+        seed ^= hasher(key.data()) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
+        auto attr_val = nostd::visit(converter, value);
```

Comment:
Why is this convert needed?

Reply:
We eventually use std::hash<t> for hash calculation. Which doesn't work directly on the nostd:: types, so need to change it to standard type before calculation.